File: Best Time to Buy and Sell Stock II/solve.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int leng = prices.size();
        int sellPrice = prices[0];
        int buyPrice = prices[0];
        int profit = 0;
        for (int i = 1; i < leng; i++)
        {
            if (sellPrice < prices[i])
            {
                sellPrice = prices[i];
            }
            else if (sellPrice > prices[i])
            {
                profit += sellPrice - buyPrice;
                buyPrice = sellPrice = prices[i];
            }
        }
        profit += (sellPrice - buyPrice);
        return profit;
    }
};

File: Best Time to Buy and Sell Stock II/solve.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    const leng = prices.length;
    let sellPrice = prices[0];
    let buyPrice = prices[0];
    let profit = 0;
    for (let i = 1; i < leng; i++) {
        if (sellPrice < prices[i]) {
            sellPrice = prices[i];
        } else if (sellPrice > prices[i]) {
            profit += sellPrice - buyPrice;
            buyPrice = sellPrice = prices[i];
        }
    }
    profit += sellPrice - buyPrice;
    return profit;
};
